include/kernel/thread.h

```diff
@@ -145,7 +145,10 @@ static inline void commitTime(void)
         }
         NODE_STATE(ksCurSC)->scConsumed += NODE_STATE(ksConsumed);
     }
-    if (CONFIG_NUM_DOMAINS > 1) {
+    /* If we have switched domains, ksDomainTime has already been reset and is accounted for the
+     * old domain. If we have not switched domains, we still need to account for the time that
+     * has passed. */
+    if (CONFIG_NUM_DOMAINS > 1 && !switchedDomain) {
         assert(ksDomainTime > NODE_STATE(ksConsumed));
         if (NODE_STATE(ksConsumed) < ksDomainTime) {
```

Comment:
Is it now possible for this if block (introduced here: 036ffc1895515a393377768124ed3d47f9b434cf) to be reverted?
It appears impossible for the else branch to be taken unless I'm missing something?

Reply:
Let's see what information we have in the proof at that point. I think you are right and it can be removed, but that optimisation should fall out of the proof. Doing it now and then re-adding it because we forgot a case would waste a lot of cycles.